content/sensu-core/0.29/reference/filters.md

```diff
@@ -295,6 +296,135 @@ value), and the event data does not contain a matching [eval token
+
+Here's an example of a handler definition that uses the occurrences filter:
+
+{{< highlight json >}}
+{
+  "handlers": {
+    "email": {
+      "type": "pipe",
+      "command": "email.rb",
+      "filters": ["occurrences"]
+    }
+  }
+}
+{{< /highlight >}}
+
+#### Occurrences filter attributes {#occurrences-filter-attributes}
+
+The occurrences filter lets you configure the number of occurrences and the refresh interval at the check level using two check definition attributes: `occurrences` and `refresh`.
+
+Here's an example of a check definition that passes events to the `email` handler starting with the third occurrence every 60 minutes:
+
+{{< highlight json >}}
+{
+  "checks": {
+    "check-http": {
+      "command": "check-http.rb -u https://localhost:8080/api/v1/health",
+      "subscribers": ["web_application"],
+      "interval": 20,
+      "handlers": ["email"],
+      "occurrences": 2,
+      "refresh": 3600
+    }
+  }
+}
+{{< /highlight >}}
+
+occurrences  | 
+-------------|------
+description  | The number of events that must occur before an event is handled for the check
+required     | True if the specified handler uses the occurrences filter
+type         | Integer
+default      | 1
+example      | {{< highlight shell >}}"occurrences": 2{{< /highlight >}}
+
+refresh      | 
+-------------|------
+description  | Time in seconds until event occurrences are handled for the check again
+required     | False
+type         | Integer
+default      | 1800
+example      | {{< highlight shell >}}"refresh": 3600{{< /highlight >}}
+
+### Check dependencies filter {#check-dependencies-filter}
+
+The check dependencies filter is included in every install of Sensu.
+This filter can be applied to a handler using the `filters` handler definition attribute.
+The check dependencies filter lets you reduce notification noise by matching events when an event already exists, thereby only notifying for the root cause of a given failure.
+
+Here's an example of a handler definition that uses the checks dependencies filter:
+
+{{< highlight json >}}
+{
+  "handlers": {
+    "custom_mailer": {
+      "type": "pipe",
+      "command": "custom_mailer.rb",
+      "filters": ["check_dependencies"]
+    }
+  }
+}
+{{< /highlight >}}
+
+#### Defining check dependencies {#defining-check-dependencies}
+
+The check dependencies filter uses a custom check definition attribute `dependencies`.
+The `dependencies` attribute should define an array containing names of checks or
+client/check pairs.
+
+Here's an example of a check definition that depends on a check named `mysql`:
+
+{{< highlight json >}}
+{
+  "checks": {
+    "web_application_api": {
+      "command": "check-http.rb -u https://localhost:8080/api/v1/health",
+      "subscribers": ["web_application"],
+      "interval": 20,
+      "dependencies": [
+        "mysql"
+      ]
+    }
+  }
+}
+{{< /highlight >}}
```

Comment:
Not sure I understand this behavior. Which event is being filtered: `mysql` or `web_application_api`?

Reply:
If `mysql` check for this client is in a non-OK state, events from this client for `web_application_api` check from will be filtered.